### PR TITLE
Upgrade Pom.xml to latest versions of Dependencies #1959

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

**Review by lmsurpre — APPROVED** [review]
LGTM

**Review by ldettman — APPROVED** [review]
Looks good.

File: fhir-parent/pom.xml [hunk 1]
```diff
@@ -101,28 +101,28 @@
...
+                <version>4.0.4</version>
                 <scope>provided</scope>
             </dependency>
             <dependency>
-                <groupId>jakarta.websocket</groupId>
-                <artifactId>jakarta.websocket-api</artifactId>
-                <version>1.1.1</version>
+                <groupId>jakarta.xml.ws</groupId>
+                <artifactId>jakarta.xml.ws-api</artifactId>
+                <version>2.3.3</version>
```
Comment (lmsurpre) [inline fhir-parent/pom.xml:125]: is this a net-new dependency?  or is this just what jakarta.websocket has become?

File: fhir-parent/pom.xml [hunk 2]
```diff
@@ -289,33 +294,44 @@
             <dependency>
                 <!-- required for the Tyrus client to work on Java 11 -->
                 <groupId>javax.xml.ws</groupId>
-                <artifactId>jaxws-api</artifactId>
-                <version>2.3.0</version>
+                <artifactId>s</artifactId>
```
Comment (lmsurpre) [inline fhir-parent/pom.xml:297]: accidental change?

File: fhir-server/pom.xml [hunk 3]
```diff
@@ -121,15 +121,14 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>javax.inject</groupId>
-            <artifactId>javax.inject</artifactId>
-            <version>1</version>
+            <groupId>jakarta.inject</groupId>
+            <artifactId>jakarta.inject-api</artifactId>
+            <version>1.0.3</version>
```
Comment (lmsurpre) [inline fhir-server/pom.xml:126]: should this version be moved to dependencyManagement too?

File: fhir-parent/pom.xml (hunk 2 above)
Comment (prb112) [reply fhir-parent/pom.xml:297]: Yes, will fix

File: fhir-server/pom.xml (hunk 3 above)
Comment (prb112) [reply fhir-server/pom.xml:126]: Yes

Comment (prb112) [reply fhir-server/pom.xml:126]: Yes! I'm not sure how I missed it.

File: fhir-parent/pom.xml (hunk 2 above)
Comment (prb112) [reply fhir-parent/pom.xml:297]: fixed

File: fhir-parent/pom.xml (hunk 1 above)
Comment (prb112) [reply fhir-parent/pom.xml:125]: It's used in fhir-server-test

Comment (prb112) [reply fhir-parent/pom.xml:126]: and it's used for fhir-server-test only with the tyrus client for websockets.

File: fhir-audit/pom.xml [hunk 4]
```diff
@@ -50,6 +50,11 @@
             <artifactId>jsonassert</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (prb112) [inline fhir-audit/pom.xml:57]: it was implied, and now that it throws a specific exception I added it

**Pull request merged**